### PR TITLE
feat: add v3 endpoints

### PR DESCRIPTION
**Pull request by NamelsKing**
- Controllers splitted on shared, v2, and v3
 - V2 controllers marked deprecated
 - Api error codes along with uniq errors (allocation, validation, etc)
 - New error filter for apiErrors

### CODE REVIEW

File: src/proofs/proofs.service.ts [hunk 1]
```diff
@@ -17,13 +20,16 @@ export class ProofsService {
   async getBouncerProof(chain: Chain, address: string): Promise<MerkleProof> {
     const fileName = `badger-bouncer-${parseInt(chain.chainId)}.json`;
     const bouncerFile = await getObject(REWARD_DATA, fileName);
+    console.log('--------------------------------------------------');
```
Comment (axejintao) [inline src/proofs/proofs.service.ts:23]: is this for test or purposeful? 
if test we can remove probably dont wanna log this each time i guess?

**Review by axejintao — APPROVED** [review]
lgtm, few comments. lmk if you want to address it later non blocking

File: src/accounts/account.v3.controller.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  @ContentType('json')
+  @Summary('Get badger user account information')
+  @Description(
+    'Return key user information for a given account. Includes positions, earnings from use, and claimable balances.',
+  )
+  @Returns(200, AccountModel)
+  @Returns(400).Description('Not a valid chain')
+  @Returns(404).Description('Not a valid account')
+  async getAccount(
+    @QueryParams('accountId') userId: string,
```
[the remark this answers is not in the document]
Comment (axejintao) [reply src/accounts/account.v3.controller.ts:24]: actually, @NamelsKing let's update this one before merge

File: src/proofs/proofs.service.ts (hunk 1 above)
Comment (NamelsKing) [reply src/proofs/proofs.service.ts:23]: gosh, my bad, will rm

File: src/accounts/accounts.v2.controller.ts [hunk 3]
```diff
@@ -21,7 +22,7 @@ export class AccountsController {
   @Returns(404).Description('Not a valid account')
   async getAccount(
     @PathParams('accountId') userId: string,
-    @QueryParams('chain') chain?: Network,
+    @QueryParams('chain') chain: Network,
```
[the remark this answers is not in the document]
Comment (NamelsKing) [reply src/accounts/accounts.v2.controller.ts:25]: it wont impact request or emth, tbh, but ye u right, its optional

**Review by axejintao — APPROVED** [review]

**Pull request merged**